### ISSUE TITLE
Cannot compile on MSVC because of min/max definition

### ISSUE THREAD

**Issue by m4nu3lf** [Zylann/godot_voxel#295]
I tried to compile Godot and the module with MSVC as I was experiencing issues with MinGW.

Compilation was failing, I had to change the name of min/max defined [here](https://github.com/Zylann/godot_voxel/blob/master/util/math/funcs.h#L37) as it seems that the Microsoft compiler is pulling in an header with a non-standard definition of min/max in the global scope that conflicts with that.

Has the module ever been compiled with MSVC before? Is it supposed to work with it?

**Comment by Zylann** [comment]
I work almost exclusively with MSVC (2017 and now 2019) and the CI also does https://github.com/Zylann/godot_voxel/blob/master/.github/workflows/windows.yml and [succeeds](https://github.com/Zylann/godot_voxel/actions/runs/1046347345).
So I'm not sure how you get this problem 🤔

**Comment by m4nu3lf** [comment]
Let me try to compile it again. It might be something to do with the fact I'm forcing Godot to compile with C++17, or something else with my setup.

**Comment by m4nu3lf** [comment]
Found the cause. It was caused by some of my code.

Basically I wrote a generator in C++ compiled with the engine. I used some STL utilities there, in particular I included `std::array` which looking at the microsoft STL implementation includes xutility which includes utility which defines min and max.

No need to change min/max then, but good to know that you might hit this issue if you use the STL in MSVC while including headers of the module.

The issue can be closed I guess.
